source/slang/slang-intrinsic-expand.cpp

```diff
@@ -223,15 +223,34 @@ const char* IntrinsicExpandContext::_emitSpecial(const char* cursor)
 
     SLANG_RELEASE_ASSERT(cursor < end);
 
-    const char d = *cursor++;
+    char d = *cursor++;
+
+    // Takes the first character of the number, parses the rest and returns the
+    // total value.
+    auto isDigit = [](char c){ return c >= '0' && c <= '9'; };
+    auto parseNat = [&](){
```

Comment:
`parseInt` ?

Reply:
Well, it doesn't support negative numbers :)